pandas/io/common.py

```diff
@@ -324,9 +325,51 @@ def _get_handle(path, mode, encoding=None, compression=None):
         else:
             f = open(path, mode)
 
+    if memory_map and hasattr(f, 'fileno'):
+        try:
+            f = MMapWrapper(f)
+        except:  # fallback: leave file handler as is
```

Comment:
@shoyer : that's not the right `NULL` - this is the one I'm referring to <a href="https://github.com/pydata/pandas/blob/master/pandas/parser.pyx#L613">here</a>.  The one you're pointing at is when all other attempts to create a source have failed.